### ISSUE TITLE
How to use an initial solution file as an initial solution

### ISSUE THREAD

**Issue by tdeenes** [reinterpretcat/vrp#72]
`--init-solution` or `-i` command line arguments can be used to provide the solver with an initial solution. However, the content of the file is not actually used as an initial solution - the initial solutions are generated dynamically. When looking at the source code, I found that the initial solution file is imported in line 368 in `vrp/vrp-cli/src/commands/solve.rs` (https://github.com/reinterpretcat/vrp/blob/5a179b68a25f36ebdef69a199f19a2731fff1471/vrp-cli/src/commands/solve.rs#L368), but it is then not even referenced if a configuration file is provided, and referenced but seemingly not used to generate the initial solution when configuration parameters are taken from the command line arguments.

Is this intended, do I misunderstand the purpose of `--init-solution`?

**Comment by reinterpretcat** [comment]
Yes, idea is to pass previous solution as initial. However, algorithm might still build more initial solutions which is controlled with `--init-size` flag. I haven't used this functionality for a while, but last time it worked as expected. Initial solutions are passed here: https://github.com/reinterpretcat/vrp/blob/5a179b68a25f36ebdef69a199f19a2731fff1471/vrp-cli/src/commands/solve.rs#L389

**Comment by tdeenes** [comment]
But if you check line 382, there is an if-else branch there, and if a config file is provided, the parsed solution is not used at all. Plus even if I do not use the config file, it is always one of the dynamically generated initial solutions which is used in the further optimization phase even though the provided initial solution is far more optimal in all criteria.

**Comment by reinterpretcat** [comment]
A, got it. Adapted a bit the code: https://github.com/reinterpretcat/vrp/commit/858b8b69410bc5ec5df9890013535ae87e081f9e

Have you tried to set `"maxSize":1` in the config file to avoid other solutions to be generated?

**Comment by tdeenes** [comment]
Yes, I found `"maxSize": 1` and set it in the config file. Now I tested your patch and indeed, it now takes into consideration the initial solution when I pass a config file, too. So far so good. And finally I think I figured out what caused the confusion: the initial costs are somehow miscalculated when the initial solution file is processed. So before your recent patch, the no-config-file version did actually use the initial solution, but its fitness values were incorrect. No the same thing happens with the config-file scenario, too.

This is what I did:
1) Call vrp-cli, store the final solution in a file. The fitness values are:
```
rank: 0, fitness: (0.000, 21.000, 706713.350)
```
Note: the target objectives are `minimize-unassigned, minimize-tours, minimize-cost`.

2) Check the final solution: it contains no unassigned jobs, but it contains 8 violations (all of them are break violations).

3) Re-run vrp-cli by passing the previous solution as an initial solution. This is what I get:
```
[0s] created 1 of 1 initial solutions in 0ms
        fitness: (8.000, 21.000, 706713.350)
[0s] generation 0 took 0ms, rank: 0, fitness: (8.000, 21.000, 706713.350), improvement: 0.000%
[0s] population state (phase: initial, speed: 0.00 gen/sec, improvement ratio: 1.000:1.000):
         rank: 0, fitness: (8.000, 21.000, 706713.350), improvement: 0.000%
slow refinement speed (0.25), switch to slower hyper-heuristic
[7s] generation 100 took 9ms, rank: 0, fitness: (0.000, 21.000, 706529.350), improvement: 0.000%
```
As you can see the first entry in the initial fitness value array is incorrect: it is 8.000 instead of 0.000. In the scenario above the optimization process quickly finds a more optimal solution, but this is not always the case. So my assumption is that the break violations are somehow considered as unassigned jobs when the initial solution is processed.

Btw, this is an amazing piece of software and I am very impressed by your responsiveness.

**Comment by tdeenes** [comment]
It seems this issue can be closed, at least I can not reproduce it with 1.16.1. Now the break violations are reflected in the fitness value even after the first run, and the second run picks the right initial solution even if less optimal alternative initial solutions are also generated.
